src/module.c

```diff
@@ -505,24 +505,34 @@ int DropIndexCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return RedisModule_ReplyWithError(ctx, "Unknown Index name");
   }
 
-  int delDocs;
-  if (RMUtil_StringEqualsCaseC(argv[0], "FT.DROP") ||
-      RMUtil_StringEqualsCaseC(argv[0], "_FT.DROP")) {
-    delDocs = 1;
-    if (argc == 3 && RMUtil_StringEqualsCaseC(argv[2], "KEEPDOCS")) {
-      delDocs = 0;
-    }
-  } else {  // FT.DROPINDEX
+  int delDocs = 0;
+  int keepDocs = 0;
+
+  if (argc == 3) {
+    if (RMUtil_StringEqualsCaseC(argv[2], "_FORCEKEEPDOCS")) {
+      keepDocs = 1;
+    } else if (RMUtil_StringEqualsCaseC(argv[0], "FT.DROP") ||
+               RMUtil_StringEqualsCaseC(argv[0], "_FT.DROP")) {
```

Comment:
Let's have this parsing outside of the `argc` condition, makes more sense.
Also try to have a single check of the 4'th argument (`DD`, `KEEPDOCS`, `_FORCEKEEPDOCS`